### PR TITLE
View refactoring

### PR DESCRIPTION
**Pull request by ldehaine**
## Changes
* Create class CardAreaOrganizer:
  * This class' work is to organize the card tiles so that ViewGame just has to iterate through 2 lists and "blit" the images at the given position
  * The idea is that the controller won't communicate directly with ViewGame when adding cards to the view, so the Organizer kinda acts like an adapter in that sense
  * The CardAreaOrganizer is able to add a card to a hand through "add_card" method and also provides a "split_hand" method that handles splitting a hand without checking the cards, though (because that's no this classe's job, and anyway card values are not stored in the organizer). Also we have a "clear_areas" method that clears all areas
* Add tests for CardAreaOrganizer
* Change how we read values in game.cfg.json and rename the file to view_game.cfg.json

### CODE REVIEW

File: data/config/game_view.cfg.json [hunk 1]
```diff
@@ -16,22 +18,17 @@
...
-                "x": 400,
-                "y": 50
-            },
-            "hand_2": {
-                "x": 800,
-                "y": 50
-            }
+        "player": {
+            "x": 200,
+            "width": 700,
```
Comment (Hboni) [inline data/config/game_view.cfg.json]: I don't like the order, can you put x and y together.

**Review by deplanty — APPROVED** [review]
:)

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
+marshmallow==2.19.5
```
Comment (Hboni) [inline requirements.txt:2]: Add dataclasses !?

File: src/views/view_game.py [hunk 3]
```diff
@@ -1,38 +1,120 @@
...
+
 from src.Button import Button
-from collections import defaultdict
+from src.CardsAPI.Card import Card
+from src.common.func_pictures import load_image, convert_card_to_picture
+from src.common.game_view_config import game_view_config, Coordinates
+from src.common.utils import Signal, SurfaceWithPosition
+
+
+class CardAreaOrganizer:
```
Comment (Hboni) [inline src/views/view_game.py]: #TODO create tests ?

File: src/views/view_game.py [hunk 4]
```diff
@@ -1,38 +1,120 @@
...
+        num_areas = len(self.player_areas)
+        player_area_width = (
+            self.player_card_area_config.width
+        )
+
+        x_pos = round(
+            self.player_card_area_config.x +
+            player_area_width / (1 + num_areas)
+        )
+        y_pos = self.player_card_area_config.y
```
Comment (Hboni) [inline src/views/view_game.py]: You don't use `x_offset` and `y_offset` :/

File: src/views/view_game.py [hunk 5]
```diff
@@ -42,36 +124,41 @@ def init_window(self):
...
-                          height=80,
-                          text='Quit',
-                          background=(180, 180, 180))
+                               width=80,
+                               height=80,
+                               text='Quit',
+                               background=(180, 180, 180))
         self.quit_btn.display(self.window)
 
     def add_card(self, card, area_name):
```
Comment (Hboni) [inline src/views/view_game.py]: Mmh, we already use this or just with the organizer ?

**Review by Hboni — CHANGES_REQUESTED** [review]

**Comment by ldehaine** [comment]
I just realized that everytime we add a new card area for the player, we will have to recalculate the position for the already placed cards... I still have some work to do

File: src/views/view_game.py (hunk 5 above)
Comment (ldehaine) [reply src/views/view_game.py]: Forgot to remove it, now we will only add_card *via* the organizer

File: src/views/view_game.py (hunk 3 above)
Comment (ldehaine) [reply src/views/view_game.py]: Absolutely

**Comment by ldehaine** [comment]
Following previous comment, I added the feature (and fixed things following @Hboni's review) but still haven't added tests. Work in progress

File: data/config/game_view.cfg.json [hunk 6]
```diff
@@ -16,22 +18,17 @@
             "y": 15
         },
         "dealer": {
-            "x": 600,
-            "y": 50
-        },
-        "player_one_hand": {
-            "x": 600,
+            "x": 550,
             "y": 50
```
Comment (Hboni) [inline data/config/game_view.cfg.json:22]: You have width for player area, do we need one for dealer ?

**Review by Hboni — APPROVED** [review]
Just one comment.

File: data/config/game_view.cfg.json (hunk 6 above)
Comment (ldehaine) [reply data/config/game_view.cfg.json:22]: We could, especially if we want to define areas to refresh independently in the game view, as we discussed last time. In that case we will need a height as well. But I'd rather keep this for another time.
For the moment it's pretty much as if the width for dealer area was 0.

**Comment by ldehaine** [comment]
We should be done now :
### New changes
* Move card_area_organizer in its own file (he was getting a bit heavy, plus, I plan on creating more organizers so they will all have their own file,
* Add split_hand method
* Add a clear_areas method

### Comments
* I have not added a way to remove a single card from an area because I don't think we would use it in the game (or I forgot something?)
* I am not very used to implementing tests so if you have any feedback or think I forgot some checks, just tell me

Also, I updated the first post to reflect all changes made by this PR

**Comment by ldehaine** [comment]
Ok I have a problem, Travis can't run the tests because of pygame (because Travis doesn't have a video device), I actually need this `pygame.display.set_mode` or pygame won't let me call `pygame.transform.scale` in `CardAreaOrganizer._create_card_tile` (stupid pygame  😠)
So I will override `_create_card_tile` in the tests so it doesn't call `pygame.transform.scale`. Anyway I really did not like having to create a window when running the tests...

**Comment by ldehaine** [comment]
Problem solved. Ok to merge?

**Review by Hboni — APPROVED** [review]

**Pull request merged**